src/mono/mono/metadata/class-setup-vtable.c

```diff
@@ -1742,11 +1742,11 @@ mono_class_setup_vtable_general (MonoClass *klass, MonoMethod **overrides, int o
 		for (int i = 0; i < iface_onum; i++) {
 			MonoMethod *decl = iface_overrides [i*2];
 			MonoMethod *override = iface_overrides [i*2 + 1];
-			if (decl->is_inflated) {
+			if (mono_class_is_gtd (override->klass)) {
+				override = mono_class_inflate_generic_method_full_checked (override, ic, mono_class_get_context (ic), error);
+			} else if (decl->is_inflated) {
 				override = mono_class_inflate_generic_method_checked (override, mono_method_get_context (decl), error);
 				mono_error_assert_ok (error);
-			} else if (mono_class_is_gtd (override->klass)) {
-				override = mono_class_inflate_generic_method_full_checked (override, ic, mono_class_get_context (ic), error);
 			}
```

Comment:
I am not certain that we need the `decl->is_inflated` case at all. I can not think of a scenario where this is needed. If we can find the testcase I would like to add that to the tests in this PR as well.

Reply:
I did change the code locally to assert if we hit the `decl->is_inflated` case and `override` changed. I then ran all the tests and I do think this this assert was hit and I got the same results. Depending on how well dims are covered with our test code, this code block may really be unnecessary.  

```
} else if (decl->is_inflated) {
	MonoMethod *toverride = mono_class_inflate_generic_method_checked (override, mono_method_get_context (decl), error);
	g_assert(toverride == override);
	override = toverride;
	mono_error_assert_ok (error);
}
```

Reply:
We'll do a separate PR for net7 
```c
} else if (decl->is_inflated) {
	g_assert_not_reached ();
}
```

we'll backport the current version to mono/mono and net6